src/alley/wp/simplecache/class-metadata-adapter.php

```diff
@@ -0,0 +1,222 @@
+	 * @return CacheInterface
+	 */
+	public static function for_term( int $term_id ): CacheInterface {
+		return self::create( 'term', $term_id );
+	}
+
+	/**
+	 * Fetches a value from the cache.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string $key     The unique key of this item in the cache.
+	 * @param mixed  $default Default value to return if the key does not exist.
+	 * @return mixed The value of the item from the cache, or $default in case of cache miss.
+	 */
+	public function get( string $key, mixed $default = null ): mixed {
+		$get = $this->getMultiple( [ $key ], $default );
+
+		$out = $default;
+
+		foreach ( $get as $value ) {
+			$out = $value;
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Persists data in the cache, uniquely referenced by a key with an optional expiration TTL time.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string                 $key   The key of the item to store.
+	 * @param mixed                  $value The value of the item to store, must be serializable.
+	 * @param null|int|\DateInterval $ttl   Optional. The TTL value of this item.
+	 * @return bool True on success and false on failure.
+	 */
+	public function set( string $key, mixed $value, \DateInterval|int|null $ttl = null ): bool {
+		return $this->setMultiple( [ $key => $value ], $ttl );
+	}
+
+	/**
+	 * Delete an item from the cache by its unique key.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string $key The unique cache key of the item to delete.
+	 * @return bool True if the item was successfully removed. False if there was an error.
+	 */
+	public function delete( string $key ): bool {
+		if ( ! $this->has( $key ) ) {
+			return true;
+		}
+
+		return (bool) delete_metadata( $this->type, $this->id, $key );
+	}
+
+	/**
+	 * Wipes clean the entire cache's keys.
+	 *
+	 * @return bool True on success and false on failure.
+	 */
+	public function clear(): bool {
+		return false; // Not supported by default in WordPress.
+	}
+
+	/**
+	 * Obtains multiple cache items by their unique keys.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If $keys is neither an array nor a Traversable, or if any of
+	 *                                                   the $keys are not a legal value.
+	 *
+	 * @phpstan-param iterable<string> $keys
+	 * @phpstan-return iterable<string, mixed>
+	 *
+	 * @param iterable $keys    A list of keys that can be obtained in a single operation.
+	 * @param mixed    $default Default value to return for keys that do not exist.
+	 * @return iterable A list of key => value pairs. Cache keys that do not exist or are stale will have $default as value.
+	 */
+	public function getMultiple( iterable $keys, mixed $default = null ): iterable {
+		$metadata = get_metadata( $this->type, $this->id );
```

Comment:
I may be misunderstanding, but does this library provide functionality for using an object cache implementation for metadata, or does it rely on the underlying WP implementation (e.g., `get_metadata` here would attempt to pull from WP's cache, which could be a memcached-backed object cache, but this library doesn't care)?

Reply:
As discussed in Slack, it's more the latter: The takeaway for this adapter is that object metadata itself is used as the cache data store. The fact that core also caches object metadata in the persistent object cache is an implementation detail on the part of core.